src/main/scala/org/broadinstitute/dsde/workbench/sam/service/UserService.scala

```diff
@@ -8,30 +8,36 @@ import javax.naming.NameNotFoundException
 import com.typesafe.scalalogging.LazyLogging
 import org.apache.commons.codec.binary.Hex
 import org.broadinstitute.dsde.workbench.model._
-import org.broadinstitute.dsde.workbench.sam.api.CreateWorkbenchUser
+import org.broadinstitute.dsde.workbench.sam.api.{CreateWorkbenchUser, InviteUser}
 import org.broadinstitute.dsde.workbench.sam.directory.DirectoryDAO
 import org.broadinstitute.dsde.workbench.sam.model._
-
 import scala.concurrent.{ExecutionContext, Future}
 
 /**
   * Created by dvoet on 7/14/17.
   */
 class UserService(val directoryDAO: DirectoryDAO, val cloudExtensions: CloudExtensions)(implicit val executionContext: ExecutionContext) extends LazyLogging {
 
-  def createUser(user: CreateWorkbenchUser): Future[UserStatus] = {
+  def createUser(user: CreateWorkbenchUser): Future[UserStatus] = for {
+      createdUser <- registerUser(user)
+      res <- provisionUser(createdUser)
+    } yield res
+
+  def inviteUser(invitee: InviteUser): Future[UserStatus] = for {
+      createdUser <- directoryDAO.createUser(WorkbenchUser(invitee.inviteeId, None, invitee.inviteeEmail))
+      res <- provisionUser(createdUser)
```

Comment:
I don't think we should provision the user when they are invited, we should do that when they register